Sources/DeckUI/Views/CameraView.swift

```diff
@@ -115,17 +154,26 @@ class ContentViewModel: ObservableObject {
 
     func prepareCamera() {
         captureSession.sessionPreset = .high
-
+        
         if let device = AVCaptureDevice.default(for: .video) {
             startSessionForDevice(device)
         }
+        
+        self.fetchDevices()
     }
 
     func startSessionForDevice(_ device: AVCaptureDevice) {
         do {
+            for input in captureSession.inputs {
+                captureSession.removeInput(input)
+            }
+
             let input = try AVCaptureDeviceInput(device: device)
+            captureSession.beginConfiguration()
             addInput(input)
```

Comment:
Probably you need to wrap this in:
`if session.canAddInput { .. }`

Reply:
Oh smart! Thank you 😊